### DISCUSSION TITLE
Can't view log for failed action

### DISCUSSION THREAD

**Discussion by cmyr** [community/community#25596] [Actions]
<p>I have <a href="https://github.com/xi-editor/druid/blob/master/.github/workflows/bloat.yml" rel="nofollow noopener">an action that runs on issue_comment</a> events. It has a single job, that is supposed to run conditionally, based on whether or not the comment begins with a magic word.</p>
<p>This was working for a while, but sometime about a week ago it broke, and now this action is showing up as a failure every time anyone comments on a PR; but more importantly I can’t view the output for these job runs. They show up in the overall ‘Actions list’, but if I click on them I am taken to a focus view (it looks like based on the commit they were triggered for?) and the failures are not listed.</p>
<p>This is mostly a problem because my collaborators get spammed with constant failure notices.</p>

**Comment by wipe2238** [comment, marked as answer]
<p>There’s no logs to see, it’s just wrong status set. <a href="https://github.community/t/status-of-workflows-with-no-running-jobs/16782">Click.</a></p>
<p>What you can do before fix, is to change your “if” in a way that tests will be done by shell, and add extra variable to “env” context if all conditions are satisfied.</p>
<pre><code>steps:
- name: Check Things
run: |
         if ["${{ github.event_name }}" == "this"] || ["something" == "else"]; then
            echo ::set-env name=RUN_JOB::true
         fi
  shell: bash
</code></pre>
<p>Next, update all other steps with extra “if” checking environment variable:</p>
<pre><code>- name: Do Things
  if: env.RUN_JOB == 'true'
  run: things.exe
</code></pre>
<p>You’ll get old. behavior (no spam if job shouldn’t run) at a price of running workflow for a second or two for each trigger. Easy to revert once we get a fix.</p>
